### PR TITLE
feat(TDI-40612): Marketo tacokit rewrite

### PR DESCRIPTION
**Pull request by undx**
https://jira.talendforge.org/browse/TDI-40612
https://jira.talendforge.org/browse/TDI-40708

### CODE REVIEW

File: marketo/src/main/java/org/talend/components/marketo/MarketoSourceOrProcessor.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        this.authorizationClient.base(this.dataSet.getDataStore().getEndpoint());
+    }
+
+    @PostConstruct
+    public void init() {
+        nextPageToken = null;
+        retrieveAccessToken();
+    }
+
+    public String getAccessToken() {
```
Comment (chmyga) [inline marketo/src/main/java/org/talend/components/marketo/MarketoSourceOrProcessor.java]: do we need this method? I could find it's usage only in one test.

File: marketo/src/main/java/org/talend/components/marketo/input/LeadSource.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    private JsonObject getLeadActivities() {
+        String sinceDateTime = getPagingToken(configuration.getSinceDateTime());
+        String activityTypeIds = "";
+        if (configuration.getActivityTypeIds().isEmpty()) {
+
+        } else {
+            activityTypeIds = configuration.getActivityTypeIds().stream().collect(joining(","));
+        }
```
[the remark this answers is not in the document]
Comment (undx) [reply marketo/src/main/java/org/talend/components/marketo/input/LeadSource.java]: Sure, done.

File: marketo/src/main/java/org/talend/components/marketo/dataset/ActivityType.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import java.io.Serializable;
+
+import org.talend.sdk.component.api.configuration.Option;
+import org.talend.sdk.component.api.configuration.action.Suggestable;
+import org.talend.sdk.component.api.meta.Documentation;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class ActivityType implements Serializable, CharSequence {
```
[the remark this answers is not in the document]
Comment (undx) [reply marketo/src/main/java/org/talend/components/marketo/dataset/ActivityType.java]: Right, done

File: marketo/src/main/java/org/talend/components/marketo/MarketoSourceOrProcessor.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        if (accessToken == null) {
+            retrieveAccessToken();
+        }
+        return accessToken;
+    }
+
+    /**
+     * Retrieve an set an access token for using API
+     */
+    public void retrieveAccessToken() {
```
[the remark this answers is not in the document]
Comment (undx) [reply marketo/src/main/java/org/talend/components/marketo/MarketoSourceOrProcessor.java]: Cleaner, done.

File: marketo/src/main/java/org/talend/components/marketo/MarketoSourceOrProcessor.java (hunk 1 above)
Comment (undx) [reply marketo/src/main/java/org/talend/components/marketo/MarketoSourceOrProcessor.java]: Right, removed.

File: marketo/src/main/java/org/talend/components/marketo/input/LeadSource.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        return handleResponse(leadClient.getLeadChanges(accessToken, nextPageToken, listId, leadIds, fields));
+    }
+
+    public JsonObject getActivities() {
+        return handleResponse(leadClient.getActivities(accessToken));
+    }
+
+    public String getPagingToken(String dateTime) {
+        Response<JsonObject> pt = leadClient.getPagingToken(accessToken, dateTime);
+        return pt.body().getString(ATTR_NEXT_PAGE_TOKEN);
```
[the remark this answers is not in the document]
Comment (undx) [reply marketo/src/main/java/org/talend/components/marketo/input/LeadSource.java]: Handling response correctly now.

**Pull request closed without merging**